src/sagemaker/model.py

```diff
@@ -407,6 +408,14 @@ def _compilation_job_config(
             else input_shape,
             "Framework": framework.upper(),
         }
+        if framework.upper() == "PYTORCH" and framework_version is not None:
+            if target_instance_type is not None:
+                if target_instance_type.startswith("ml_"):
+                    if "ml_inf" not in target_instance_type:
+                        input_model_config["FrameworkVersion"] = utils.get_short_version(
+                            framework_version
+                        )
```

Comment:
[Flat is better than nested](https://www.python.org/dev/peps/pep-0020/):

```suggestion
        if (
            framework.lower() == "pytorch" and
            framework_version is not None and
            target_instance_type is not None and
            target_instance_type.startswith("ml_") and
            "ml_inf" not in target_instance_type
        ):
            input_model_config["FrameworkVersion"] = utils.get_short_version(framework_version)
```

If you imported `re`, you could also provide a regex for the last two. Something like:

```
re.match("(?=^ml_)(?!ml_inf)", target_instance_type) is not None
```

Reply:
Thanks for the review!! I have updated the changes accordingly.